timegate/config.py

```diff
@@ -41,33 +43,60 @@ def from_inifile(self, filename, silent=True):
             self['API_TIME_OUT'] = conf.getfloat('server', 'api_time_out')
 
         # Handler configuration
-        if conf.has_option('handler', 'handler_class'):
-            self['HANDLER_MODULE'] = conf.get('handler', 'handler_class')
-        if conf.has_option('handler', 'base_uri'):
-            self['BASE_URI'] = conf.get('handler', 'base_uri')
-        if conf.getboolean('handler', 'is_vcs'):
-            self['RESOURCE_TYPE'] = 'vcs'
-        else:
-            self['RESOURCE_TYPE'] = 'snapshot'
-
-        if conf.has_option('handler', 'use_timemap'):
-            self['USE_TIMEMAPS'] = conf.getboolean('handler', 'use_timemap')
-        else:
-            self['USE_TIMEMAPS'] = False
+        def build_handler(section):
+            """Build handler configuration."""
+            output = {}
+            if conf.has_option(section, 'handler_class'):
+                output['HANDLER_MODULE'] = conf.get(section, 'handler_class')
+            if conf.has_option(section, 'base_uri'):
+                output['BASE_URI'] = conf.get(section, 'base_uri')
+            if conf.getboolean(section, 'is_vcs'):
+                output['RESOURCE_TYPE'] = 'vcs'
+            else:
+                output['RESOURCE_TYPE'] = 'snapshot'
+
+            if conf.has_option(section, 'use_timemap'):
+                output['USE_TIMEMAPS'] = conf.getboolean(section,
+                                                         'use_timemap')
+            else:
+                output['USE_TIMEMAPS'] = False
+            return output
+
+        self.setdefault('HANDLERS', {})
+        for section_name in conf.sections():
+            if section_name == 'handler' or \
+                    section_name.startswith('handler:'):
+                section = build_handler(section_name)
+                if ':' in section_name:
+                    self['HANDLERS'][section_name[len('handler:'):]] = section
+                else:
+                    self.update(section)
```

Comment:
I have improved the config handling. The "old" config values are used as defaults for all handlers. One can still keep default values and override the "default" (`None`) hander:

```python
HANDLERS = {None: {'USE_TIMEMAP': False}}
```
or
```ini
[handler]
use_timemap = False

[handler:]  # see the "trick" with :
use_timemap = True

[handler:simple]
# falls back to use_timemap = False
```